packages/cadl-ranch-specs/http/type/model/inheritance/nested-discriminator/main.tsp

```diff
@@ -0,0 +1,223 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Illustrates multiple level inheritance with multiple discriminators.")
+@scenarioService("/type/model/inheritance/nested-discriminator")
+namespace Type.Model.Inheritance.NestedDiscriminator;
+
+@doc("This is base model for polymorphic multiple levels inheritance with a discriminator.")
+@discriminator("kind")
+model Fish {
+  age: int32;
+}
+
+@doc("The second level model in polymorphic multiple levels inheritance and it defines a new discriminator.")
+@discriminator("sharktype")
```

Comment:
@archerzz my understanding was csharp wasn't able to handle multiple discriminators in one inheritance line https://github.com/Azure/typespec-azure/issues/2219

Reply:
they are not but this is still good to keep this test, we just separated it so it doesn't prevent the others from being generated.

That way we have clear documentation that this is not a feature supported in all languages instead of having to figure out why its working sometimes in the future when someone ask for it and we forgot.

Reply:
correct. that's one of the initiatives of the pr. we split the nested discriminator scenario from the original scenario, so that in csharp generator test cases we can select those scenarios we support.